### ISSUE TITLE
[MX] Support mixed MXFP4/FP6/FP8 linear layer

### ISSUE THREAD

**Issue by balancap** [pytorch/ao#1666]
Blackwell hardware natively supports any combination of MXFP4/FP6/FP8 in matmuls. See PTX and Cutlass documentation:
* https://docs.nvidia.com/cuda/parallel-thread-execution/#tcgen05-kind-shapes 
* https://github.com/NVIDIA/cutlass/blob/main/tools/library/src/reference/block_scaled_gemm_mixed8bitsa.cu

According the [MX paper](https://arxiv.org/abs/2310.10537), and macore generally the large quantization literature, there is advantages to use different bitwidth for weights, activations and gradients. It would be very useful in `mx_mm` and `MXLinear` to support this more general setting.

**Comment by balancap** [comment]
#1667 is adding mixed element dtypes support to `mx_mm`. A similar general interface could be added to `MXLinear`
```python
class MXLinear(torch.nn.Linear):
    @classmethod
    @torch.no_grad()
    def from_float(cls, mod, in_elem_dtype, w_elem_dtype, grad_elem_dtype, block_size):
        ...
```

**Comment by jainapurva** [comment]
@vkuzo @drisspg

**Comment by balancap** [comment]
Closing the issue as #1667 has been merged
